src/lib/paymentOptions/NimiqPaymentOptions.ts

```diff
@@ -0,0 +1,186 @@
+        }
+
+        let feePerByte: number | undefined;
+        if (options.protocolSpecific.feePerByte !== undefined) {
+            try {
+                feePerByte = parseFloat(toNonScientificNumberString(options.protocolSpecific.feePerByte));
+            } catch (e) {
+                throw new Error('If provided, feePerByte must be a valid number');
+            }
+        }
+
+        let fee: number | undefined;
+        if (options.protocolSpecific.fee !== undefined) {
+            if (!this.isNonNegativeInteger(options.protocolSpecific.fee)) {
+                throw new Error('If provided, fee must be a non-negative integer');
+            }
+            fee = Number.parseInt(toNonScientificNumberString(options.protocolSpecific.fee), 10);
+        }
+
+        const requiresExtendedTransaction = extraData.length > 0
+            || recipientType !== undefined && recipientType !== Nimiq.Account.Type.BASIC
+            || flags !== undefined && flags !== Nimiq.Transaction.Flag.NONE;
+        // Note that the transaction size can be bigger than this, for example if the sender type the user wants to use
+        // requires an extended transaction or if an extended transaction includes a multi signature proof. The size
+        // is therefore just an estimate. In the majority of cases the estimate will be accurate though and a fee that
+        // is slightly off will generally not be a problem.
+        const estimatedTransactionSize = requiresExtendedTransaction ? 166 + extraData.length : 138;
+
+        if (fee === undefined) {
+            if (feePerByte === undefined) {
+                feePerByte = 0;
+                fee = 0;
+            } else {
+                fee = Math.ceil(feePerByte * estimatedTransactionSize);
+            }
+        } else {
+            feePerByte = fee / estimatedTransactionSize;
+        }
+
+        if (options.protocolSpecific.validityDuration !== undefined
+            && !this.isNonNegativeInteger(options.protocolSpecific.validityDuration)) {
+            throw new Error('If provided, validityDuration must be a non-negative integer.');
+        }
+
+        this.protocolSpecific = {
+            sender,
+            forceSender: !!options.protocolSpecific.forceSender,
+            fee,
+            feePerByte,
+            flags: flags || Nimiq.Transaction.Flag.NONE,
+            recipient,
+            recipientType,
+            validityDuration: !options.protocolSpecific.validityDuration
+                ? TX_VALIDITY_WINDOW
+                : Math.min(
+                    TX_VALIDITY_WINDOW,
+                    Math.max(
+                        TX_MIN_VALIDITY_DURATION,
+                        options.protocolSpecific.validityDuration,
+                    ),
+                ),
+        };
+    }
+
+    public get currency(): Currency.NIM {
+        return Currency.NIM;
+    }
+
+    public get type(): PaymentMethod.DIRECT {
+        return PaymentMethod.DIRECT;
```

Comment:
As a general question: Is this only static right now because it's the only type we support and will be made dynamic in the future, or will OASIS payments need a new PaymentOptions class?